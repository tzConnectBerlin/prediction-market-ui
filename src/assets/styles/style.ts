import { css, SerializedStyles } from '@emotion/react';
import { Theme } from '@material-ui/core';

export const GlobalStyle = (theme: Theme): SerializedStyles => css`
  body {
    margin: 0;
    font-family: 'Roboto', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }
  .MuiAlert-standardError {
    color: ${theme.palette.error.main};
    background-color: ${theme.palette.error.dark};
  }
  .MuiAlert-standardSuccess {
    color: ${theme.palette.success.main};
    background-color: ${theme.palette.success.dark};
  }
  .MuiAlert-standardWarning {
    color: ${theme.palette.warning.main};
    background-color: ${theme.palette.warning.dark};
  }
  .MuiFormControl-root {
    margin-top: ${theme.spacing(1)};
    margin-bottom: ${theme.spacing(1)};
    width: 100%;
    button[disabled],
    .MuiButton-root.Mui-disabled {
      background-color: ${theme.palette.primary.main};
      color: ${theme.palette.buttonText.disabled};
    }
    .MuiFormHelperText-root {
      color: ${theme.palette.text.secondary};
      &:not(.extra-help-message) {
        color: ${theme.palette.warning.main};
      }
    }
    .MuiInput-root.MuiInputBase-formControl {
      &:before,
      &.Mui-disabled::before,
      &:hover:not(.Mui-disabled)::before,
      &:after {
        border-bottom: none;
      }
      background-color: ${theme.palette.grey[400]};
      border: solid 1px ${theme.palette.grey[600]};
      border-radius: ${theme.spacing(1 / 4)};
      padding-left: ${theme.spacing(1 / 2)};
      padding-right: ${theme.spacing(1 / 2)};
      &:not(.Mui-disabled) {
        &:hover {
          background-color: ${theme.palette.grey[500]};
        }
        &:focus {
          border-color: ${theme.palette.primary.main};
        }
      }
      &.Mui-error {
        border-color: ${theme.palette.warning.main};
      }
      .MuiSelect-select {
        padding-left: ${theme.spacing(1 / 2)};
      }
      .MuiSelect-icon {
        color: ${theme.palette.text.primary};
      }
    }
  }
  .MuiFormLabel-root.MuiInputLabel-root,
  .MuiInputLabel-formControl.MuiInputLabel-root {
    font-weight: bold;
    color: ${theme.palette.primary.main};
    position: relative;
    transform: none;
    transition: none;
    font-size: 0.8rem;
    &.Mui-disabled {
      color: ${theme.palette.primary.main};
      opacity: 0.38;
    }
    & .label-asterisk {
      vertical-align: sub;
      line-height: 0;
    }
  }
  .MuiSlider {
    &-root {
      color: ${theme.palette.primary.main};
      &.Mui-disabled {
        color: ${theme.palette.primary.main};
        opacity: 0.38;
      }
    }
    &-thumb {
      background-color: ${theme.palette.primary.main};
      &.Mui-disabled {
        background-color: ${theme.palette.primary.main};
      }
    }
  }
`;
