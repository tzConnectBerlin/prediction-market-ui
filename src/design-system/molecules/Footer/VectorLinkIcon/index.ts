export { VectorLinkIcon, VectorLinkIcon as default } from './VectorLinkIcon';
export type { VectorLinkIconProps } from './VectorLinkIcon';
