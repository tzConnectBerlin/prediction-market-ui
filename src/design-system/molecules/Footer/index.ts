export { Footer as default, Footer } from './Footer';
export type { Props } from './Footer';
