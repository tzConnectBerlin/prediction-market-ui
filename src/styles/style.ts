import { css, SerializedStyles } from '@emotion/react';
import { Theme } from '@mui/material';

export const GlobalStyle = (theme: Theme): SerializedStyles => css`
  @import url('https://fonts.googleapis.com/css2?family=Roboto:wght@400;700&display=swap');
  @import url('https://fonts.googleapis.com/css2?family=Roboto+Mono:wght@400;700&display=swap');
  body {
    margin: 0;
    font-family: 'Roboto', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    background-color: ${theme.palette.background.default};
  }
  .pointer {
    cursor: pointer;
  }
  .MuiPaper-root {
    box-shadow: 0 0 3px ${theme.palette.grey[600]};
  }
  button[disabled],
  .MuiButton-root.Mui-disabled {
    &:not(.MuiToggleButton-root, .MuiPickersDay-root, .MuiIconButton-edgeEnd, .MuiTab-root) {
      background-color: ${theme.palette.primary.main};
      color: ${theme.palette.buttonText.disabled};
    }
  }

  .MuiButtonBase-root.MuiIconButton-root {
    &:hover {
      background-color: transparent;
    }
  }

  .MuiFormControl-root {
    margin-top: ${theme.spacing(1)};
    margin-bottom: ${theme.spacing(1)};
    width: 100%;
    .MuiFormHelperText-root {
      color: ${theme.palette.text.secondary};
      &:not(.extra-help-message) {
        color: ${theme.palette.warning.main};
      }
    }
    .MuiInput-root.MuiInputBase-formControl {
      background-color: ${theme.palette.grey[300]};
      border: solid 1px ${theme.palette.grey[600]};
      border-radius: ${theme.spacing(1 / 4)};

      &:hover:not(.Mui-disabled) {
        background-color: ${theme.palette.grey[500]};
      }
      &.Mui-focused:not(.Mui-disabled) {
        border-color: ${theme.palette.primary.main};
      }
      &.Mui-error {
        border-color: ${theme.palette.warning.main};
      }

      &:before,
      &.Mui-disabled::before,
      &:hover:not(.Mui-disabled)::before,
      &:after {
        border-bottom: none;
      }
      background-color: ${theme.palette.grey[300]};
      border: solid 1px ${theme.palette.grey[600]};
      border-radius: ${theme.spacing(1 / 4)};
      padding-left: ${theme.spacing(1 / 2)};
      padding-right: ${theme.spacing(1 / 2)};
      input,
      textarea {
        padding-left: ${theme.spacing(1 / 2)};
        padding-right: ${theme.spacing(1 / 2)};
      }
      &:not(.Mui-disabled) {
        &:hover {
          background-color: ${theme.palette.grey[500]};
        }
        &:focus {
          border-color: ${theme.palette.primary.main};
        }
      }
      &.Mui-error {
        border-color: ${theme.palette.warning.main};
      }
      .MuiSelect-select {
        padding-left: ${theme.spacing(1 / 2)};
      }
      .MuiSelect-icon {
        color: ${theme.palette.text.primary};
      }
    }
  }
`;
