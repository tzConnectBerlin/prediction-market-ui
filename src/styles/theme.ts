import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(255, 255, 255, 1)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    light: 'rgba(1, 102, 255, 0.6)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.06)',
    '300': 'rgba(29, 34, 39, 0.04)',
    '200': 'rgba(29, 34, 39, 0.03)',
    '100': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.65)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.65)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
  typography: {
    fontWeightRegular: 400,
    fontWeightBold: 700,
    h1: {
      fontWeight: 700,
      fontSize: '1.875rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h2: {
      fontWeight: 700,
      fontSize: '1.125rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
    h3: {
      fontWeight: 700,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h4: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h5: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    h6: {
      fontWeight: 700,
      fontSize: '0.625rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    subtitle1: {
      fontWeight: 700,
      fontSize: '0.75rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto Mono',
    },
    body1: {
      fontWeight: 400,
      fontSize: '1rem',
      lineHeight: '1.3',
      fontFamily: 'Roboto',
    },
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 900,
      lg: 1200,
      xl: 1536,
    },
  },
  zIndex: {
    tooltip: 9999,
  },
});
