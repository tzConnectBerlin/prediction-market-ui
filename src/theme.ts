import { createTheme, PaletteOptions } from '@material-ui/core/styles';

const lightThemePalette: PaletteOptions = {
  primary: {
    dark: 'rgba(26, 117, 255, 1)',
    main: 'rgba(1, 102, 255, 1)',
    light: 'rgba(1, 102, 255, 0.12)',
  },
  secondary: {
    dark: 'rgba(1, 102, 255, 0.10)',
    main: 'rgba(1, 102, 255, 0.12)',
    contrastText: 'rgba(1, 102, 255, 1)',
  },
  success: {
    dark: 'rgba(212, 241, 204, 1)',
    main: 'rgba(42, 184, 0, 1)',
    light: 'rgba(212, 241, 204, 0.12)',
  },
  warning: {
    dark: 'rgba(255, 229, 212, 1)',
    main: 'rgba(253, 125, 42, 1)',
  },
  error: {
    dark: 'rgba(255, 215, 215, 1)',
    main: 'rgba(255, 57, 57, 1)',
    light: 'rgba(255, 215, 215, 0.12)',
  },
  tertiary: {
    dark: 'rgba(105, 255, 209, 1)',
    main: 'rgba(57, 255, 194, 1)',
    light: 'rgba(57, 255, 194, 0.12)',
    contrastText: 'rgba(29, 34, 39, 0.87)',
  },
  grey: {
    '900': 'rgba(29, 34, 39, 1)',
    '800': 'rgba(29, 34, 39, 0.87)',
    '700': 'rgba(29, 34, 39, 0.38)',
    '600': 'rgba(29, 34, 39, 0.09)',
    '500': 'rgba(29, 34, 39, 0.07)',
    '400': 'rgba(29, 34, 39, 0.04)',
    '300': 'rgba(29, 34, 39, 0.03)',
    '200': 'rgba(29, 34, 39, 0.02)',
  },
  text: {
    primary: 'rgba(29, 34, 39, 0.87)',
    secondary: 'rgba(29, 34, 39, 0.54)',
    disabled: 'rgba(29, 34, 39, 0.38)',
  },
  buttonText: {
    primary: 'rgba(255, 255, 255, 0.87)',
    secondary: 'rgba(255, 255, 255, 0.54)',
    disabled: 'rgba(255, 255, 255, 0.38)',
  },
  background: {
    default: 'rgba(253, 253, 255, 1)',
    paper: 'rgba(255, 255, 255, 1)',
  },
  tonalOffset: 0.05,
  mode: 'light',
  // action: {
  //   disabledBackground: "#0166FF",
  //   disabled: "rgba(255,255,255,0.3)",
  // },
};

export const lightTheme = createTheme({
  palette: lightThemePalette,
});

export const darkTheme = createTheme({
  palette: {
    ...lightThemePalette,
    primary: {
      dark: 'rgba(214, 231, 255, 1)',
      main: 'rgba(204, 224, 255, 1)',
      light: 'rrgba(204, 224, 255, .12)',
      contrastText: 'rgba(29, 34, 39, 0.87)',
    },
    secondary: {
      dark: 'rgba(204, 224, 255, 0.14)',
      main: 'rgba(204, 224, 255, 0.12)',
      light: 'rgba(204, 224, 255, 0.14)',
      contrastText: 'rgba(204, 224, 255, 1)',
    },
    tertiary: {
      dark: 'rgba(212, 241, 204, 1)',
      main: 'rgba(212, 241, 204, 1)',
      light: 'rgba(212, 241, 204, 0.12)',
    },
    grey: {
      '900': 'rgba(255, 255, 255, 1)',
      '800': 'rgba(255, 255, 255, 0.87)',
      '700': 'rgba(255, 255, 255, 0.38)',
      '600': 'rgba(255, 255, 255, 0.2)',
      '500': 'rgba(255, 255, 255, 0.11)',
      '400': 'rgba(255, 255, 255, 0.09)',
      '300': 'rgba(255, 255, 255, 0.03)',
      '200': 'rgba(255, 255, 255, 0.02)',
    },
    text: {
      primary: 'rgba(255, 255, 255, 0.87)',
      secondary: 'rgba(255, 255, 255, 0.54)',
      disabled: 'rgba(255, 255, 255, 0.38)',
    },
    buttonText: {
      primary: 'rgba(29, 34, 39, 0.87)',
      secondary: 'rgba(29, 34, 39, 0.54)',
      disabled: 'rgba(29, 34, 39, 0.38)',
    },
    background: {
      default: 'rgba(29, 34, 39, 1)',
      paper: 'rgba(45, 49, 54, 1)',
    },
    mode: 'dark',
  },
  zIndex: {
    tooltip: 9999,
  },
});
