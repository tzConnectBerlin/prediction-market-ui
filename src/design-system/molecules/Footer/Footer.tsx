import React from 'react';
import styled from '@emotion/styled';
import { Grid, Box, Container } from '@material-ui/core';
import { Typography } from '../../atoms/Typography';
import { VectorLinkIcon } from './VectorLinkIcon';

const FooterStyled = styled.footer`
  margin-top: auto;
`;

const FooterContainer = styled(Container)`
  width: 100%;
  display: flex;
  margin-top: 2rem;
  position: relative;
  top: auto;
  bottom: 0;
  padding: 2.5rem;
`;

const LinkTypographyStyled = styled(Typography)`
  cursor: pointer;
  padding-left: 0;
  padding-bottom: 2rem;
  font-weight: normal !important;
`;

interface FooterLink {
  label: string;
  handleLinkClick?: () => void | Promise<void>;
  isExternal?: boolean;
}
export interface FooterProps {
  title?: string;
  links?: FooterLink[];
  description: string[];
}

export const Footer: React.FC<FooterProps> = ({
  /**
   * List of descriptions
   */
  description,
  /**
   * header for description of tezos prediction markets
   */
  title,
  /**
   * list of allowed actions
   */
  links = [],
}) => {
  return (
    <FooterStyled>
      <FooterContainer
        sx={{ backgroundColor: 'background.default' }}
        disableGutters
        maxWidth={false}
      >
        <Grid container direction="column" flexWrap="nowrap">
          {title && (
            <Grid item xs={12}>
              <Typography color="text.primary" size="h6">
                {title}
              </Typography>
            </Grid>
          )}
          <Grid container item xs={12} md={4} direction="row">
            {links.map(({ label, isExternal, handleLinkClick }, index) => {
              return (
                <Grid
                  item
                  xs={6}
                  sm={6}
                  md={4}
                  lg={2}
                  justifyContent="flex-start"
                  key={`${label}-${index}`}
                >
                  <LinkTypographyStyled
                    color="primary.main"
                    size="h3"
                    onClick={handleLinkClick}
                    whiteSpace="nowrap"
                  >
                    {label}
                    {isExternal && (
                      <Box marginLeft="0.3rem" component="span">
                        <VectorLinkIcon />
                      </Box>
                    )}
                  </LinkTypographyStyled>
                </Grid>
              );
            })}
          </Grid>
          <Grid container item xs={12} md={8} lg={9}>
            <Grid container item direction="column">
              {description.map((content, index) => (
                <Grid item key={index}>
                  <Typography color="text.secondary" size="body1" marginY={0.5}>
                    {content}
                  </Typography>
                </Grid>
              ))}
            </Grid>
          </Grid>
        </Grid>
      </FooterContainer>
    </FooterStyled>
  );
};
