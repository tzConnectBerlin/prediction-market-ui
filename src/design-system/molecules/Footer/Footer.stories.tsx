import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { ThemeProvider } from '@material-ui/core';
import { lightTheme as theme } from '../../../theme';
import { Footer, Props } from './Footer';

export default {
  title: 'Molecule/Footer',
  component: Footer,
} as Meta;

const Template: Story<Props> = (args) => (
  <ThemeProvider theme={theme}>
    <Footer {...args} />
  </ThemeProvider>
);

export const WithTitleAndText = Template.bind({});
WithTitleAndText.args = {
  title: 'About Tezos Prediction Markets',
  footerDescriptionFirst:
    'This software implements a crowd-funded prediction market. Once the market has been initiated by setting the question and other details, other participants can give their own assessments of the likelihood of the question coming true, and investing in the market. Not only will these initial participants have a stake in the market, they will also profit from its operation, taking a share of the market’s fees.',
  footerDescriptionSecond:
    'From then on the market operates in a standard way–each token will pay out to one or zero coins,depending upon which way events transpire. A 5% fee on winnings rewards participants in the investment phase, and those who have contributed liquidity to the market during its operation.',
  footerLinkHow: 'HOW IT WORKS',
  footerLinkAbout: 'ABOUT TEZOS',
};
