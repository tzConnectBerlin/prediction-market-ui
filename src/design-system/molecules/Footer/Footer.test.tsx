import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@mui/material';
import { Footer, FooterProps } from './Footer';
import { lightTheme as theme } from '../../../styles/theme';

const defaultProps: FooterProps = {
  title: 'About Tezos Prediction Markets',
  description: [
    'This software implements a crowd-funded prediction market. Once the market has been initiated by setting the question and other details, other participants can give their own assessments of the likelihood of the question coming true, and investing in the market. Not only will these initial participants have a stake in the market, they will also profit from its operation, taking a share of the market’s fees.',
    'From then on the market operates in a standard way–each token will pay out to one or zero coins,depending upon which way events transpire. A 5% fee on winnings rewards participants in the investment phase, and those who have contributed liquidity to the market during its operation.',
  ],
  links: [
    {
      label: 'HOW IT WORKS',
    },
    {
      label: 'ABOUT TEZOS',
      isExternal: true,
    },
  ],
};

const WrappedComponent: React.FC<any> = (props) => (
  <ThemeProvider theme={theme}>
    <Footer {...props} />
  </ThemeProvider>
);

describe('Snapshot testing Footer Component', () => {
  it('renders correctly with Props', () => {
    const footer = renderer.create(<WrappedComponent {...defaultProps} />).toJSON();
    expect(footer).toMatchSnapshot();
  });
});

describe('Element testing Header Component', () => {
  it('render correctly on LoggedOut and title', async () => {
    const { getByText } = render(<WrappedComponent {...defaultProps} />);

    expect(getByText(/About Tezos Prediction Markets/i)).toBeInTheDocument();
    expect(
      getByText(/This software implements a crowd-funded prediction market/i),
    ).toBeInTheDocument();
  });
});
