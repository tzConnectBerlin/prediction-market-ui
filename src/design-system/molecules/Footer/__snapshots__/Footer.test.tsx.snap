// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing Footer Component renders correctly with Props 1`] = `
.emotion-0 {
  margin-top: auto;
}

.emotion-1 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  background-color: #fff;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 2rem;
  position: relative;
  top: auto;
  bottom: 0;
  padding: 2.5rem;
}

.emotion-2 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: nowrap;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.emotion-2>.MuiGrid-item {
  max-width: none;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-4.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-5 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
}

@media (min-width:900px) {
  .emotion-5 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

@media (min-width:1200px) {
  .emotion-5 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 50%;
  -ms-flex-preferred-size: 50%;
  flex-basis: 50%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 50%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
}

@media (min-width:600px) {
  .emotion-6 {
    -webkit-flex-basis: 50%;
    -ms-flex-preferred-size: 50%;
    flex-basis: 50%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 50%;
  }
}

@media (min-width:900px) {
  .emotion-6 {
    -webkit-flex-basis: 33.333333%;
    -ms-flex-preferred-size: 33.333333%;
    flex-basis: 33.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 33.333333%;
  }
}

@media (min-width:1200px) {
  .emotion-6 {
    -webkit-flex-basis: 16.666667%;
    -ms-flex-preferred-size: 16.666667%;
    flex-basis: 16.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 16.666667%;
  }
}

@media (min-width:1536px) {
  .emotion-6 {
    -webkit-flex-basis: 8.333333%;
    -ms-flex-preferred-size: 8.333333%;
    flex-basis: 8.333333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 8.333333%;
  }
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 3rem;
  line-height: 1.167;
  letter-spacing: 0em;
  color: #1976d2;
  white-space: nowrap;
  cursor: pointer;
  padding-left: 0;
  padding-bottom: 2rem;
  font-weight: normal!important;
}

.emotion-7.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

.emotion-10 {
  margin-left: 0.3rem;
}

.emotion-11 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
}

@media (min-width:900px) {
  .emotion-11 {
    -webkit-flex-basis: 66.666667%;
    -ms-flex-preferred-size: 66.666667%;
    flex-basis: 66.666667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 66.666667%;
  }
}

@media (min-width:1200px) {
  .emotion-11 {
    -webkit-flex-basis: 75%;
    -ms-flex-preferred-size: 75%;
    flex-basis: 75%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 75%;
  }
}

.emotion-12 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  margin: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-12>.MuiGrid-item {
  max-width: none;
}

.emotion-13 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-14 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.6);
  margin-top: 4px;
  margin-bottom: 4px;
}

.emotion-14.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<footer
  className="emotion-0"
>
  <div
    className="MuiContainer-root MuiContainer-disableGutters emotion-1"
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-undefined MuiGrid-direction-xs-column emotion-2"
    >
      <div
        className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-3"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6  emotion-4"
          lines={3}
        >
          About Tezos Prediction Markets
        </h6>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-undefined MuiGrid-grid-xs-12 MuiGrid-grid-md-4 MuiGrid-grid-lg-3 emotion-5"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2 MuiGrid-grid-xl-1 emotion-6"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3 emotion-7"
            lines={3}
          >
            HOW IT WORKS
          </h3>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 MuiGrid-grid-lg-2 MuiGrid-grid-xl-1 emotion-6"
        >
          <h3
            className="MuiTypography-root MuiTypography-h3 emotion-7"
            lines={3}
          >
            ABOUT TEZOS
            <span
              className="MuiBox-root emotion-10"
            >
              <svg
                height={12}
                viewBox="0 0 10 12"
                width={10}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M9.96461 1.05338L10.0002 7.88981C10.0002 8.15426 9.80879 8.36974 9.56838 8.36974C9.32798 8.36974 9.13654 8.15916 9.13654 7.89471L9.10538 2.16014L0.744591 11.357C0.575416 11.548 0.299395 11.548 0.13022 11.357C-0.0434067 11.1709 -0.0434067 10.8722 0.13022 10.6812L8.49101 1.48433L3.27776 1.45495C3.03735 1.45005 2.84592 1.23947 2.84592 0.975024C2.84592 0.710578 3.0418 0.5 3.28221 0.5L9.49715 0.539177C9.78208 0.539177 9.95125 0.749755 9.96461 1.05338Z"
                  fill="#0E61F6"
                />
              </svg>
            </span>
          </h3>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-undefined MuiGrid-grid-xs-12 MuiGrid-grid-md-8 MuiGrid-grid-lg-9 emotion-11"
      >
        <div
          className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-undefined MuiGrid-direction-xs-column emotion-12"
        >
          <div
            className="MuiGrid-root MuiGrid-item emotion-13"
          >
            <p
              className="MuiTypography-root MuiTypography-body1  emotion-14"
              lines={3}
            >
              This software implements a crowd-funded prediction market. Once the market has been initiated by setting the question and other details, other participants can give their own assessments of the likelihood of the question coming true, and investing in the market. Not only will these initial participants have a stake in the market, they will also profit from its operation, taking a share of the marketâ€™s fees.
            </p>
          </div>
          <div
            className="MuiGrid-root MuiGrid-item emotion-13"
          >
            <p
              className="MuiTypography-root MuiTypography-body1  emotion-14"
              lines={3}
            >
              From then on the market operates in a standard wayâ€“each token will pay out to one or zero coins,depending upon which way events transpire. A 5% fee on winnings rewards participants in the investment phase, and those who have contributed liquidity to the market during its operation.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</footer>
`;
